kview/src/test/java/dev/ikm/komet/kview/state/test/ConceptDetailsStateMachineTest.java

```diff
@@ -0,0 +1,86 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.ikm.komet.kview.state.test;
+
+import dev.ikm.komet.kview.state.pattern.ConceptDetailsPattern;
+import org.carlfx.axonic.StateMachine;
+import org.carlfx.axonic.tools.StateMachineCLI;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static dev.ikm.komet.kview.state.ConceptDetailsState.BF_NO_FQN;
+import static dev.ikm.komet.kview.state.ConceptDetailsState.EDITING_OTN_NO_FQN;
+import static org.carlfx.axonic.tools.DiagramHelper.toPlantUml;
+
+@DisplayName("StateBuilder Test")
+public class ConceptDetailsStateMachineTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ConceptDetailsStateMachineTest.class);
+
+    @Test
+    @DisplayName("Test Concept Details state machine UI flow happy path.")
+    void conceptDetailsStateMachineFlowTest() {
+        StateMachine sm = StateMachine.create(new ConceptDetailsPattern());
+
+        String expected = """
+                name: initial from: INITIAL to: BF_NO_FQN
+                name: toggle press from: BF_NO_FQN to: BF_NO_FQN
+                name: edit Otn from: BF_NO_FQN to: EDITING_OTN_NO_FQN
+                name: add Fqn from: BF_NO_FQN to: ADDING_FQN
+                name: add Otn from: BF_NO_FQN to: ADDING_OTN_NO_FQN
+                name: done from: ADDING_OTN_NO_FQN to: BF_NO_FQN
+                name: cancel from: ADDING_OTN_NO_FQN to: BF_NO_FQN
+                name: cancel from: ADDING_FQN to: BF_NO_FQN
+                name: done from: ADDING_FQN to: BF_WITH_FQN
+                name: add Otn from: BF_WITH_FQN to: ADDING_OTN_WITH_FQN
+                name: cancel from: ADDING_OTN_WITH_FQN to: BF_WITH_FQN
+                name: done from: ADDING_OTN_WITH_FQN to: BF_WITH_FQN
+                name: cancel from: EDITING_FQN to: BF_WITH_FQN
+                name: done from: EDITING_FQN to: BF_WITH_FQN
+                name: edit fqn from: BF_WITH_FQN to: EDITING_FQN
+                name: done from: EDITING_OTN_NO_FQN to: BF_NO_FQN
+                name: cancel from: EDITING_OTN_NO_FQN to: BF_NO_FQN
+                name: edit Otn from: BF_WITH_FQN to: EDITING_OTN_WITH_FQN
+                name: done from: EDITING_OTN_WITH_FQN to: BF_WITH_FQN
+                name: cancel from: EDITING_OTN_WITH_FQN to: BF_WITH_FQN
+                """;
+        StringBuilder sb = new StringBuilder();
+        sm.getStatePattern().transitions().forEach(transition -> {
+            sb.append("name: %s from: %s to: %s\n".formatted(transition.name(), transition.fromState(), transition.toState()));
+        });
+        System.out.println(sb);
```

Comment:
Not a big deal, but did you want to leave this println in here?

Reply:
Oh darn. Missed it, I pressed merge. Oh well, I'll remove later. It's in a test anyways.